src/transformers/models/pop2piano/feature_extraction_pop2piano.py

```diff
@@ -0,0 +1,300 @@
+            Step between each window of the waveform.
+        frequency_min (`float`, *optional*, defaults to 10.0):
+            Minimum frequency.
+        nb_mel_filters (`int`, *optional*, defaults to 512):
+            Number of Mel filers to generate.
+        n_bars (`int`, *optional*, defaults to 2):
+            Determines `n_steps` in method `preprocess_mel`. Per `n_step` th beat is taken from each sequence.
+    """
+    model_input_names = ["input_features"]
+
+    def __init__(
+        self,
+        sampling_rate: int = 22050,
+        use_mel: int = True,
+        padding_value: int = 0,
+        fft_window_size: int = 4096,
+        hop_length: int = 1024,
+        frequency_min: float = 10.0,
+        nb_mel_filters: int = 512,
+        n_bars: int = 2,
+        feature_size=None,
+        **kwargs,
+    ):
+        super().__init__(
+            feature_size=feature_size,
+            sampling_rate=sampling_rate,
+            padding_value=padding_value,
+            **kwargs,
+        )
+        self.n_bars = n_bars
+        self.sampling_rate = sampling_rate
+        self.use_mel = use_mel
+        self.padding_value = padding_value
+        self.fft_window_size = fft_window_size
+        self.hop_length = hop_length
+        self.frequency_min = frequency_min
+        self.nb_mel_filters = nb_mel_filters
+
+    def log_mel_spectogram(self, sequence):
+        """Generates MelSpectrogram then applies log base e."""
+
+        mel_fb = get_mel_filter_banks(
+            nb_frequency_bins=(self.fft_window_size // 2) + 1,
+            nb_mel_filters=self.nb_mel_filters,
+            frequency_min=self.frequency_min,
+            frequency_max=float(self.sampling_rate // 2),
+            sample_rate=self.sampling_rate,
+            norm=None,
+            mel_scale="htk",
+        ).astype(np.float32)
+
+        spectrogram = []
+        for seq in sequence:
+            window = np.hanning(self.fft_window_size + 1)[:-1]
+            framed_audio = fram_wave(seq, self.hop_length, self.fft_window_size)
+            spec = stft(framed_audio, window, fft_window_size=self.fft_window_size)
+            spec = np.abs(spec) ** 2.0
+            spectrogram.append(spec)
+
+        spec_shape = spec.shape
+        spectrogram = torch.Tensor(spectrogram).view(-1, *spec_shape)
+        log_melspec = (
+            torch.matmul(spectrogram.transpose(-1, -2), torch.from_numpy(mel_fb))
+            .transpose(-1, -2)
+            .clamp(min=1e-6)
+            .log()
+        )
+
+        return log_melspec
+
+    def extract_rhythm(self, raw_audio):
+        """
+        This algorithm(`RhythmExtractor2013`) extracts the beat positions and estimates their confidence as well as
+        tempo in bpm for an audio signal. For more information please visit
+        https://essentia.upf.edu/reference/std_RhythmExtractor2013.html .
+        """
+        essentia_tracker = essentia.standard.RhythmExtractor2013(method="multifeature")
```

Comment:
Presumably too difficult / time consuming to port this Essentia C++ algorithm to pure Numpy? https://pypi.org/project/essentia/ We can keep this as an additional dependency if so

Reply:
I am sorry right now I don't have the time to convert the Essentia C++ algorithm to pure numpy, but I can guarantee that next month I can have a look at it. How about we ignore it for now, and I  will update  it with a new PR later?